patroni/postmaster.py

```diff
@@ -154,9 +155,19 @@ def start(pgcommand, data_dir, conf, options):
         except psutil.NoSuchProcess:
             pass
 
+        # Windows doesn't support close_fds.
+        # Warning: The preexec_fn parameter is not safe to use in the presence of threads in your application. 
+        # The child process could deadlock before exec is called. If you must use it, keep it trivial! 
+        # Minimize the number of libraries you call into.
+        # If you need to modify the environment for the child use the env parameter rather than doing it in a preexec_fn. 
+        # The start_new_session parameter can take the place of a previously common use of preexec_fn to call 
+        # os.setsid() in the child. 
         proc = call_self(['pg_ctl_start', pgcommand, '-D', data_dir,
-                          '--config-file={}'.format(conf)] + options, close_fds=True,
-                         preexec_fn=os.setsid, stdout=subprocess.PIPE, env=env)
+                          '--config-file={}'.format(conf)] + options,
+                          close_fds=True if os.name != 'nt' else False,         
+                           _fn=os.setsid if os.name != 'nt' else None,    
```

Comment:
`preexec_fn=os.setsid` was here from the `pg_ctl` times, now we can move it to https://github.com/zalando/patroni/blob/master/patroni/__init__.py#L155-L158

```python
def pg_ctl_start(args):
    import subprocess
    if os.name != 'nt':
        os.setsid()
    postmaster = subprocess.Popen(args)
    print(postmaster.pid)
```

Reply:
And remove copy&paste from docs.python.org ;)